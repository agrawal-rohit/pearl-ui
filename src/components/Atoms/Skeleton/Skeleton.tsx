// import React from "react";
// import {
//   backgroundColor,
//   BackgroundColorProps,
//   color,
//   ColorProps,
//   layout,
//   LayoutProps,
//   spacing,
//   SpacingProps,
//   opacity,
//   OpacityProps,
//   visible,
//   VisibleProps,
// } from "../../../theme/src/styleFunctions";
// import { RNStyle, StyleFunctionContainer } from "../../../theme/src/types";
// import {
//   AntDesign,
//   Entypo,
//   EvilIcons,
//   Feather,
//   FontAwesome,
//   FontAwesome5,
//   Fontisto,
//   Foundation,
//   Ionicons,
//   MaterialCommunityIcons,
//   MaterialIcons,
//   Octicons,
//   SimpleLineIcons,
//   Zocial,
// } from "@expo/vector-icons";
// import { useAtomicComponentConfig } from "../../../hooks/useAtomicComponentConfig";
// import responsiveSize from "../../../utils/responsiveSize";

// const iconStyleFunctions = [
//   color,
//   backgroundColor,
//   spacing,
//   layout,
//   opacity,
//   visible,
// ] as StyleFunctionContainer[];

// type IconProps = ColorProps &
//   BackgroundColorProps &
//   SpacingProps &
//   LayoutProps &
//   OpacityProps &
//   VisibleProps & {
//     /** Icon family that contains the icon you want to use  */
//     iconFamily:
//       | "AntDesign"
//       | "Entypo"
//       | "EvilIcons"
//       | "Feather"
//       | "FontAwesome"
//       | "FontAwesome5"
//       | "Fontisto"
//       | "Foundation"
//       | "Ionicons"
//       | "MaterialCommunityIcons"
//       | "MaterialIcons"
//       | "Octicons"
//       | "SimpleLineIcons"
//       | "Zocial";
//     /** Name of the icon as given in it's respective icon family */
//     iconName: string;
//     /** The size of the icon */
//     size?: string;
//     /** The variant of the icon */
//     variant?: string;
//     /** The accessibility label of the icon */
//     accessibilityLabel?: string;
//     style?: RNStyle;
//   };

// const iconFamilyMapping = {
//   AntDesign,
//   Entypo,
//   EvilIcons,
//   Feather,
//   FontAwesome,
//   FontAwesome5,
//   Fontisto,
//   Foundation,
//   Ionicons,
//   MaterialCommunityIcons,
//   MaterialIcons,
//   Octicons,
//   SimpleLineIcons,
//   Zocial,
// };

// /** he `Icon` component can used to add Expo Icons to your app and customize them using style props. */
// const Icon: React.FC<IconProps> = ({
//   iconFamily,
//   iconName,
//   size = "m",
//   accessibilityLabel = null,
//   ...rest
// }) => {
//   const props = useAtomicComponentConfig(
//     "Skeleton",
//     rest,
//     {
//       size: size,
//       variant: rest.variant,
//     },
//     iconStyleFunctions
//   );

//   const IconToUse = iconFamilyMapping[iconFamily];

//   return (
//     <IconToUse
//       accessible={true}
//       accessibilityLabel={
//         accessibilityLabel ? accessibilityLabel : `${iconName} Icon`
//       }
//       name={iconName}
//       {...props}
//       size={responsiveSize(props.size)}
//     ></IconToUse>
//   );
// };

// export default Icon;
