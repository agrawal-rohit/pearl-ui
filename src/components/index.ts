// Atoms
export { default as Box, BoxProps } from "./atoms/box/box";
export { default as Center } from "./atoms/center/center";
export { default as Collapse, CollapseProps } from "./atoms/collapse/collapse";
export { default as Divider, DividerProps } from "./atoms/divider/divider";
export { default as Fade, FadeProps } from "./atoms/fade/fade";
export { default as Grid, GridProps } from "./atoms/grid/grid";
export { default as Icon, IconProps } from "./atoms/icon/icon";
export {
  default as Pressable,
  PressableProps,
} from "./atoms/pressable/pressable";
export {
  default as ScaleFade,
  ScaleFadeProps,
} from "./atoms/scale-fade/scale-fade";
export {
  default as List,
  OrderedList,
  UnorderedList,
  ListProps,
} from "./atoms/list/list";
export { default as Screen, ScreenProps } from "./atoms/screen/screen";
export { default as Skeleton, SkeletonProps } from "./atoms/skeleton/skeleton";
export {
  default as SkeletonCircle,
  SkeletonCircleProps,
} from "./atoms/skeleton/skeleton-circle";
export {
  default as SkeletonText,
  SkeletonTextProps,
} from "./atoms/skeleton/skeleton-text";
export { default as Slide, SlideProps } from "./atoms/slide/slide";
export {
  default as SlideFade,
  SlideFadeProps,
} from "./atoms/slide-fade/slide-fade";
export { default as Spacer } from "./atoms/spacer/spacer";
export { default as Spinner, SpinnerProps } from "./atoms/spinner/spinner";
export {
  default as Stack,
  HStack,
  VStack,
  ZStack,
  HStackProps,
  VStackProps,
  StackProps,
} from "./atoms/stack/stack";
export { default as Text, TextProps } from "./atoms/text/text";

// Molecules
export { default as Avatar, AvatarProps } from "./molecules/avatar/avatar";
export {
  default as AvatarGroup,
  AvatarGroupProps,
} from "./molecules/avatar/avatar-group";
export { default as Badge, BadgeProps } from "./molecules/badge/badge";
export {
  default as withBadge,
  WithBadgeOptions,
} from "./molecules/badge/withBadge";
export { default as Button, ButtonProps } from "./molecules/button/button";
export {
  default as ButtonGroup,
  ButtonGroupProps,
} from "./molecules/button/button-group";
export {
  default as CheckBox,
  CheckBoxProps,
} from "./molecules/checkbox/checkbox";
export {
  default as CheckBoxGroup,
  CheckBoxGroupProps,
} from "./molecules/checkbox/checkbox-group";
export {
  default as IconButton,
  IconButtonProps,
} from "./molecules/icon-button/icon-button";
export { default as Image, ImageProps } from "./molecules/image/image";
export { default as Input, InputProps } from "./molecules/input/input";
export {
  default as PinInput,
  PinInputProps,
} from "./molecules/pin-input/pin-input";
export {
  default as Progress,
  ProgressProps,
} from "./molecules/progress/progress";
export { default as Radio, RadioProps } from "./molecules/radio/radio";
export {
  default as RadioGroup,
  RadioGroupProps,
} from "./molecules/radio/radio-group";
export { default as Switch, SwitchProps } from "./molecules/switch/switch";
export {
  default as Textarea,
  TextareaProps,
} from "./molecules/textarea/textarea";
export { default as Video, VideoProps } from "./molecules/video/video";
export {
  default as Accordion,
  AccordionProps,
  AccordionItemProps,
  AccordionButtonProps,
  AccordionPanelProps,
  AccordionItem,
  AccordionButton,
  AccordionPanel,
} from "./molecules/accordion/accordion";
export { default as Alert, AlertProps } from "./molecules/alert/alert";
