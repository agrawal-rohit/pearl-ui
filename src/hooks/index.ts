export { useColorModeValue } from "./useColorModeValue";
export { useComponentConfig } from "./useComponentConfig";
export { useStyledProps } from "./useStyledProps";
export { useTheme } from "./useTheme";
