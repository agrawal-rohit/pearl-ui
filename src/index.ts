export { default as Box } from "./components/Box/Box";
export { default as Button } from "./components/Button/Button";
export { default as Text } from "./components/Text/Text";
