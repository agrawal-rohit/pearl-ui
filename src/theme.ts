import { createTheme } from "@shopify/restyle";

const palette = {
  purpleLight: "#8C6FF7",
  purplePrimary: "#5A31F4",
  purpleDark: "#3F22AB",

  greenLight: "#56DCBA",
  greenPrimary: "#0ECD9D",
  greenDark: "#0A906E",

  black: "#0B0B0B",
  white: "#F0F2F3",
};

const theme = createTheme({
  colors: {
    primary: palette.purplePrimary,
    ...palette,
  },
  spacing: {
    s: 8,
    m: 16,
    l: 24,
    xl: 40,
  },
  breakpoints: {
    phone: 0,
    tablet: 768,
  },
});

export type Theme = typeof theme;
export default theme;
