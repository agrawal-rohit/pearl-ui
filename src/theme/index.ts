// Theme functions
export { createTheme, extendTheme } from "./src/themeFunctions";

// Style functions
export {
  color,
  opacity,
  backgroundColor,
  spacing,
  typography,
  layout,
  position,
  border,
  shadow,
  textShadow,
} from "./src/styleFunctions";

// Theme Context
export { ThemeProvider } from "./src/themeContext";
export { baseTheme } from "./src/basetheme";

// Utils
export { generatePalette } from "./utils/utils";

// Types
export { BasePearlTheme as PearlTheme } from "./src/types";
