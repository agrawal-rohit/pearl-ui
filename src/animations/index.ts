import { useAnimationState } from "../hooks/useAnimationState";

/**
 * Custom hook for creating a fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from an opacity of 0 to an opacity of 1.
 * This creates a fade-in effect.
 *
 * @returns An animation state that creates a fade-in effect.
 */
export const useFade = () => {
  // Define the initial and final states for the fade animation.
  // The animation will start with an opacity of 0 (completely transparent) and end with an opacity of 1 (completely opaque).
  const fadeAnimationState = useAnimationState({
    from: {
      opacity: 0,
    },
    to: {
      opacity: 1,
    },
  });

  // Return the fade animation state.
  return fadeAnimationState;
};

/**
 * Custom hook for creating a scale and fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a scale of 0.5 and opacity of 0 to a scale of 1 and opacity of 1.
 * This creates a combined effect of scaling and fading in.
 *
 * @returns An animation state that creates a scale and fade-in effect.
 */
export const useScaleFade = () => {
  // Define the initial and final states for the scale and fade animation.
  // The animation will start with a scale of 0.5 (half the original size) and opacity of 0 (completely transparent)
  // and end with a scale of 1 (original size) and opacity of 1 (completely opaque).
  const scaleFadeAnimationState = useAnimationState({
    from: {
      scale: 0.9,
      opacity: 0,
    },
    to: {
      scale: 1,
      opacity: 1,
    },
  });

  // Return the scale and fade animation state.
  return scaleFadeAnimationState;
};

/**
 * Custom hook for creating a slide animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a translateX of -100 to a translateX of 0.
 * This creates a slide-in effect.
 *
 * @returns An animation state that creates a slide-in effect.
 */
export const useSlide = () => {
  // Define the initial and final states for the slide animation.
  // The animation will start with a translateX of -100 (completely off-screen to the left) and end with a translateX of 0 (original position).
  const slideAnimationState = useAnimationState({
    from: {
      translateY: -100,
    },
    to: {
      translateY: 0,
    },
  });

  // Return the slide animation state.
  return slideAnimationState;
};

/**
 * Custom hook for creating a slide and fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a translateX of -100 and opacity of 0 to a translateX of 0 and opacity of 1.
 * This creates a combined effect of sliding and fading in.
 *
 * @returns An animation state that creates a slide and fade-in effect.
 */
export const useSlideFade = () => {
  // Define the initial and final states for the slide and fade animation.
  // The animation will start with a translateX of -100 (completely off-screen to the left) and opacity of 0 (completely transparent)
  // and end with a translateX of 0 (original position) and opacity of 1 (completely opaque).
  const slideFadeAnimationState = useAnimationState({
    from: {
      opacity: 0,
      translateY: 10,
    },
    to: {
      opacity: 1,
      translateY: 0,
    },
  });

  // Return the slide and fade animation state.
  return slideFadeAnimationState;
};

/**
 * Custom hook for creating a collapse animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a scaleY of 1 to a scaleY of 0.
 * This creates a collapse effect.
 *
 * @returns An animation state that creates a collapse effect.
 */
export const useCollapse = () => {
  // Define the initial and final states for the collapse animation.
  // The animation will start with a scaleY of 1 (original size) and end with a scaleY of 0 (collapsed).
  const collapseAnimationState = useAnimationState({
    from: {
      opacity: 0,
      height: 0,
    },
    to: {
      opacity: 1,
      height: 100,
    },
  });

  // Return the collapse animation state.
  return collapseAnimationState;
};
