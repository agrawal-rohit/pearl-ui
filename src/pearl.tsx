import {
  AtomComponent,
  ComponentTypes,
  FinalPearlTheme,
  MoleculeComponent,
  PearlComponent,
  StyleFunctionContainer,
} from "./theme/src/types";
import React from "react";
import { useAtomicComponentConfig } from "./hooks/useAtomicComponentConfig";
import { boxStyleFunctions, BoxStyleProps } from "./theme/src/style-functions";
import { useStyleProps } from "./hooks/useStyleProps";
import { useMolecularComponentConfig } from "./hooks/useMolecularComponentConfig";
import { MotiView, motify } from "moti";
import { useMotiWithStyleProps } from "./hooks/useMotiWithStyleProps";
import { useTheme } from "./hooks/useTheme";
import { Platform } from "react-native";

/**
 * Configuration object that adds metadata to a component that's required by Pearl UI
 */
interface pearlConfig {
  /**
   * The name of the component in the FinalPearlTheme
   */
  componentName: keyof FinalPearlTheme["components"];
  /**
   * The type of the component
   */
  type: ComponentTypes;
  /**
   * Whether the component is animatable or not
   */
  animatable?: boolean;
}

/**
 * Hook to convert a third-party component to a Pearl component that can be configured using style props, an atomic component configuration or a molecular component configuration.
 * @param Component The base component on which the pearlification has to be applied
 * @param config A configuration object that adds metadata to a component that's required by Pearl UI
 * @param styleFunctions The set of style functions that would define the style props that should be supported by the pearlified component
 * @param moleculeConfigOptions The options for configuring the molecular component
 * @returns
 */
export function pearl<
  ComponentProps,
  ComponentType extends ComponentTypes = "basic",
  ComponentAtoms extends Record<string, any> = Record<string, any>,
  StyleProps = BoxStyleProps,
  Animateable extends boolean = true,
>(
  Component: any,
  config: pearlConfig = {
    componentName: "None",
    type: "basic",
    animatable: true,
  },
  styleFunctions: StyleFunctionContainer[] = boxStyleFunctions,
  moleculeConfigOptions: {
    partForOverridenStyleProps?: string | undefined;
    partForOverridenNativeProps?: string | undefined;
    partForOverridenAnimationProps?: string | undefined;
  } = {}
) {
  let FinalComponent: any | undefined;
  const isAnimatable = !!config.animatable && config.type !== "molecule";

  if (isAnimatable) {
    const ConvertedClassComponent = class extends React.Component<any, any> {
      static displayName = `Pearl${
        Component.name ?? Component.displayName ?? `NoName`
      }`;

      render() {
        const { children, ...props } = this.props;
        return (
          <Component ref={(props as any).forwardedRef} {...props}>
            {children}
          </Component>
        );
      }
    };

    if (Platform.OS === "web" && config.componentName === "Box")
      FinalComponent = MotiView;
    else FinalComponent = motify(ConvertedClassComponent)();
  } else {
    FinalComponent = Component;
  }

  return React.memo(
    React.forwardRef(
      (
        {
          children,
          ...rest
        }: PearlComponent<ComponentProps, StyleProps, Animateable> &
          AtomComponent<(typeof config)["componentName"], ComponentType> &
          MoleculeComponent<(typeof config)["componentName"], ComponentType> & {
            atoms?: Partial<ComponentAtoms>;
            children?: string | JSX.Element | JSX.Element[] | React.ReactNode;
          },
        ref: any
      ) => {
        const { colorMode } = useTheme();

        let convertedProps;
        if (config.type === "atom") {
          convertedProps = useAtomicComponentConfig(
            config["componentName"],
            rest,
            {
              size: (rest as any).size,
              variant: (rest as any).variant,
            },
            rest.colorScheme,
            styleFunctions
          );
          if (config.animatable)
            convertedProps = useMotiWithStyleProps(
              convertedProps,
              styleFunctions
            );
        } else if (config.type === "molecule") {
          convertedProps = useMolecularComponentConfig(
            config["componentName"],
            rest,
            {
              size: (rest as any).size,
              variant: (rest as any).variant,
            },
            rest.colorScheme,
            styleFunctions,
            moleculeConfigOptions.partForOverridenStyleProps,
            moleculeConfigOptions.partForOverridenNativeProps,
            moleculeConfigOptions.partForOverridenAnimationProps
          );
        } else {
          convertedProps = useStyleProps(rest, styleFunctions);
          if (config.animatable) {
            convertedProps = useMotiWithStyleProps(
              convertedProps,
              styleFunctions
            );
          }
        }

        return (
          <FinalComponent
            {...convertedProps}
            key={
              Platform.OS === "web" &&
              ["Text", "Icon", "Box"].includes(config.componentName as string)
                ? (convertedProps as any).key!
                  ? `${(convertedProps as any).key!}-${colorMode}`
                  : `${colorMode}`
                : undefined
            }
            ref={ref}
          >
            {children}
          </FinalComponent>
        );
      }
    )
  );
}
