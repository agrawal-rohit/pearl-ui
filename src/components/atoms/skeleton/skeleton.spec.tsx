import React from "react";
import Skeleton from "./skeleton";
import SkeletonCircle from "./skeleton-circle";
import SkeletonText from "./skeleton-text";
import { render } from "@testing-library/react-native";
import { ThemeProvider } from "../../../theme/src/theme-context";
import { Text } from "react-native";

jest.useFakeTimers();

describe("Atoms/Skeleton", () => {
  it("passes the snapshot test", () => {
    const component = render(
      <ThemeProvider>
        <Skeleton>
          <Text>Test</Text>
        </Skeleton>
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });

  it("should render with default props", () => {
    const component = render(
      <ThemeProvider>
        <Skeleton>
          <Text>Test</Text>
        </Skeleton>
      </ThemeProvider>
    );
    expect(component).toBeTruthy();
  });

  it("should render with custom props", () => {
    const component = render(
      <ThemeProvider>
        <Skeleton
          startColor="neutral.100"
          endColor="neutral.200"
          isLoaded={false}
          speed={800}
          fadeDuration={200}
        >
          <Text>Test</Text>
        </Skeleton>
      </ThemeProvider>
    );
    expect(component).toBeTruthy();
  });
});

describe("Atoms/SkeletonCircle", () => {
  it("passes the snapshot test", () => {
    const component = render(
      <ThemeProvider>
        <SkeletonCircle boxSize={20} />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});

describe("Atoms/SkeletonText", () => {
  it("passes the snapshot test", () => {
    const component = render(
      <ThemeProvider>
        <SkeletonText noOfLines={3} skeletonHeight={15} spacing="3">
          <Text>Test</Text>
        </SkeletonText>
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
