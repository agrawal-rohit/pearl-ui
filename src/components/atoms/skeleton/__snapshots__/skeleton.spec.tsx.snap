// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Skeleton passes the snapshot test 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={
      {
        "backgroundColor": "#E4E9F2",
        "borderRadius": 8,
      }
    }
    animatedStyle={
      {
        "value": {
          "backgroundColor": "#F7F9FC",
        },
      }
    }
    collapsable={false}
    from={
      {
        "backgroundColor": "#F7F9FC",
      }
    }
    style={
      {
        "backgroundColor": "#F7F9FC",
        "borderRadius": 8,
      }
    }
    transition={
      {
        "duration": 800,
        "loop": true,
        "repeatReverse": true,
      }
    }
  >
    <View
      style={
        {
          "opacity": 0,
        }
      }
    >
      <Text>
        Test
      </Text>
    </View>
  </View>
</View>
`;

exports[`Atoms/SkeletonCircle passes the snapshot test 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={
      {
        "backgroundColor": "#E4E9F2",
        "borderRadius": 1000,
        "height": 20,
        "width": 20,
      }
    }
    animatedStyle={
      {
        "value": {
          "backgroundColor": "#F7F9FC",
        },
      }
    }
    collapsable={false}
    from={
      {
        "backgroundColor": "#F7F9FC",
      }
    }
    style={
      {
        "backgroundColor": "#F7F9FC",
        "borderRadius": 1000,
        "height": 20,
        "width": 20,
      }
    }
    transition={
      {
        "duration": 800,
        "loop": true,
        "repeatReverse": true,
      }
    }
  >
    <View
      style={
        {
          "opacity": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Atoms/SkeletonText passes the snapshot test 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={
      {
        "width": "100%",
      }
    }
    animatedStyle={
      {
        "value": {
          "width": "100%",
        },
      }
    }
    collapsable={false}
    style={
      {
        "flexDirection": "column",
        "flexWrap": "nowrap",
        "width": "100%",
      }
    }
  >
    <View
      animate={
        {
          "backgroundColor": "#E4E9F2",
          "borderRadius": 8,
          "height": 15,
          "marginBottom": 12,
          "width": "100%",
        }
      }
      animatedStyle={
        {
          "value": {
            "backgroundColor": "#F7F9FC",
          },
        }
      }
      collapsable={false}
      from={
        {
          "backgroundColor": "#F7F9FC",
        }
      }
      style={
        {
          "backgroundColor": "#F7F9FC",
          "borderRadius": 8,
          "height": 15,
          "marginBottom": 12,
          "width": "100%",
        }
      }
      transition={
        {
          "duration": 800,
          "loop": true,
          "repeatReverse": true,
        }
      }
    >
      <View
        style={
          {
            "opacity": 0,
          }
        }
      >
        <Text>
          Test
        </Text>
      </View>
    </View>
    <View
      animate={
        {
          "backgroundColor": "#E4E9F2",
          "borderRadius": 8,
          "height": 15,
          "marginBottom": 12,
          "width": "100%",
        }
      }
      animatedStyle={
        {
          "value": {
            "backgroundColor": "#F7F9FC",
          },
        }
      }
      collapsable={false}
      from={
        {
          "backgroundColor": "#F7F9FC",
        }
      }
      style={
        {
          "backgroundColor": "#F7F9FC",
          "borderRadius": 8,
          "height": 15,
          "marginBottom": 12,
          "width": "100%",
        }
      }
      transition={
        {
          "duration": 800,
          "loop": true,
          "repeatReverse": true,
        }
      }
    >
      <View
        style={
          {
            "opacity": 0,
          }
        }
      />
    </View>
    <View
      animate={
        {
          "backgroundColor": "#E4E9F2",
          "borderRadius": 8,
          "height": 15,
          "width": "100%",
        }
      }
      animatedStyle={
        {
          "value": {
            "backgroundColor": "#F7F9FC",
          },
        }
      }
      collapsable={false}
      from={
        {
          "backgroundColor": "#F7F9FC",
        }
      }
      style={
        {
          "backgroundColor": "#F7F9FC",
          "borderRadius": 8,
          "height": 15,
          "width": "100%",
        }
      }
      transition={
        {
          "duration": 800,
          "loop": true,
          "repeatReverse": true,
        }
      }
    >
      <View
        style={
          {
            "opacity": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
